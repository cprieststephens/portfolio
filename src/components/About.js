import React from "react";
import logo from "../img/logo.png";
import "../styles/about.css";

const About = () => {
  return (
    <div className="about">
      <img className="about__logo" src={logo} alt="laptop logo" />
      <h1 className="about__title">Camilla Priest-Stephens</h1>
      <h2 className="about__title">Associate Developer</h2>
      <p className="about__text">
        I am a former Speech and Language Therapist, career changer, graduate of
        Command Shift bootcamp and mentee on the Cajigo 100 Women in Tech
        Accelerator programme.
      </p>
      <p className="about__text">
        I created this portfolio to dislay some of my frontend projects, which
        were built with React, JavaScript, CSS, Jest and React Testing Library.
      </p>
    </div>
  );
};

export default About;
