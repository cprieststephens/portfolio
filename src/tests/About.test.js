import React from "react";
import { render, screen } from "@testing-library/react";
import About from "../components/About";

describe("About", () => {
  it("renders correctly", () => {
    const { asFragment } = render(<About />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders the correct title", () => {
    render(<About />);
    const title = screen.getByText(/camilla priest-stephens/i);

    expect(title).toBeInTheDocument();
  });

  it("renders the correct image", () => {
    render(<About />);
    const logo = screen.getByAltText("laptop logo");

    expect(logo).toBeInTheDocument();
  });
});
