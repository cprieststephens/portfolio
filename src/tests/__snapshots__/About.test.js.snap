// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About renders correctly 1`] = `
<DocumentFragment>
  <div
    class="about"
  >
    <img
      alt="laptop logo"
      class="about__logo"
      src="logo.png"
    />
    <h1
      class="about__title"
    >
      Camilla Priest-Stephens
    </h1>
    <h2
      class="about__title"
    >
      Frontend Developer
    </h2>
    <p
      class="about__text"
    >
      I am a graduate of Command Shift bootcamp looking for a role as a Frontend Developer.
    </p>
    <p
      class="about__text"
    >
      I have experience of React, JavaScript, CSS, Jest and React Testing Library and am always open to learning new tech!
    </p>
  </div>
</DocumentFragment>
`;
